langchain/vectorstores/redisearch.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import json
+import uuid
+from typing import Any, Callable, Iterable, List, Mapping, Optional
+
+import numpy as np
+from redis.commands.search.query import Query
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.utils import get_from_dict_or_env
+from langchain.vectorstores.base import VectorStore
+
+
+class RediSearch(VectorStore):
+    def __init__(
+        self, redisearch_url: str, index_name: str, embedding_function: Callable
+    ):
+        """Initialize with necessary components."""
+        try:
+            import redis
+        except ImportError:
+            raise ValueError(
+                "Could not import redis python package. "
+                "Please install it with `pip install redis`."
+            )
+        self.embedding_function = embedding_function
+        self.index_name = index_name
+        try:
+            pool = redis.ConnectionPool.from_url(redisearch_url)
```

Comment:
Might leave open ``**kwargs`` for connection pool args https://redis.readthedocs.io/en/stable/connections.html#connection-pools